app/forms/employment/EmploymentStartDateForm.scala

```diff
@@ -35,34 +30,47 @@ object EmploymentStartDateForm {
-
-  def employmentStartDateForm: Form[EmploymentDate] = Form(
-    mapping(
-      startDateDay -> trimmedText.verifying(dayEmpty andThen isDayNumeric) ,
-      startDateMonth -> trimmedText.verifying(monthEmpty andThen isMonthNumeric),
-      startDateYear -> trimmedText.verifying(yearEmpty andThen isYearNumeric)
-    )(EmploymentDate.apply)(EmploymentDate.unapply)
-  )
+  def employmentStartDateForm: Form[EmploymentDate] =
+    Form(
+      mapping(
+        startDateDay -> trimmedText,
+        startDateMonth -> trimmedText,
+        startDateYear -> trimmedText
+      )(EmploymentDate.apply)(EmploymentDate.unapply)
+    )
 
   def fifthAprilDate(taxYear: Int): LocalDate = LocalDate.of(taxYear, 4, 5)
 
-  def verifyNewDate(date: EmploymentDate, taxYear: Int): Either[FormError, LocalDate] = {
-    val newDate: Either[Throwable, LocalDate] = Try(LocalDate.of(date.amountYear.toInt, date.amountMonth.toInt, date.amountDay.toInt)).toEither
-    newDate match {
-      case Right(date) =>
-        (date.isAfter(LocalDate.now()), date.isBefore(fifthAprilDate(taxYear))) match {
-          case (true, _) => Left(FormError("invalidFormat", "employment.employmentStartDate.error.notInPast"))
-          case (_, false) => Left(FormError("invalidFormat", "employment.employmentStartDate.error.tooRecent", Seq(taxYear.toString)))
-          case _ => Right(date)
+  def areDatesEmpty(date: EmploymentDate, isAgent: Boolean): Seq[FormError] = {
+    val agentCheck: String = if (isAgent) "agent" else "individual"
+    (date.amountDay.isEmpty, date.amountMonth.isEmpty, date.amountYear.isEmpty) match{
+      case (true, true, true) => Seq(FormError(s"emptyAll", s"employment.employmentStartDate.error.incompleteAll.$agentCheck"))
+      case (true, true, false) => Seq(FormError("emptyDayMonth", s"employment.employmentStartDate.error.incompleteDayMonth.$agentCheck"))
+      case (true, false, true) => Seq(FormError("emptyDayYear", s"employment.employmentStartDate.error.incompleteDayYear.$agentCheck"))
+      case (false, true, true) => Seq(FormError("emptyMonthYear", s"employment.employmentStartDate.error.incompleteMonthYear.$agentCheck"))
+      case (false, false, true) => Seq(FormError("emptyYear", s"employment.employmentStartDate.error.incompleteYear.$agentCheck"))
+      case (false, true, false) => Seq(FormError("emptyMonth", s"employment.employmentStartDate.error.incompleteMonth.$agentCheck"))
+      case (true, false, false) => Seq(FormError("emptyDay", s"employment.employmentStartDate.error.incompleteDay.$agentCheck"))
+      case (false, false, false) => Seq()
+    }
+  }
+
+  def verifyNewDate(date: EmploymentDate, taxYear: Int, isAgent: Boolean): Seq[FormError] = {
+    val agentCheck = if (isAgent) "agent" else "individual"
+    val areDatesEmptyCall = areDatesEmpty(date, isAgent)
+    areDatesEmptyCall.isEmpty match {
+      case true =>
```

Comment:
where areDateEmptyCall.isEmpty is true it reads as if it should be false. I'm not sure if the name should be changed as the logic makes sense

Reply:
renamed to hopefully make it more readable